app/assets/scripts/views/analytics-admin-area.js

```diff
@@ -56,80 +68,110 @@ var AnalyticsAA = React.createClass({
     );
   },
 
+  // before mount, get the admin info needed to make the list of child elements
+  // as well as build the correct api queries in getAdminData
   componentWillMount: function () {
-    this.props._fetchVProMMsidsProperties();
-    this.props._fetchFieldVProMMsids();
-    this.props._fetchAdminChildren(this.props.params.aaId);
+    this.props._fetchAdminInfo(this.props.params.aaId);
+    this.props._setCrossWalk();
   },
 
-  handlePop: function () {
-    this.props.adminChildren.level === 'district' ? this.props._fetchAdminChildren(this.props.adminChildren.parent) : '';
+  // on each unmount, drain properties and admin info objects so to
+  // make the component to be the same each time
+  componentWillUnmount: function () {
+    this.props._removeAdminVProMMsProps();
+    this.props._removeAdminInfo();
   },
 
-  renderAnalyticsAdmin: function () {
-    this.props.location.action === 'POP' ? this.handlePop() : '';
-    const level = this.props.adminChildren.level;
-    const provinceId = (level === 'province') ? _.invert(this.props.crosswalk[level])[this.props.params.aaId] : this.props.crosswalk[level][this.props.params.aaId];
-    const provinceName = this.props.adminChildren.name;
-    const idTest = (level === 'province') ? new RegExp(provinceId) : [new RegExp(provinceId), new RegExp(_.invert(this.props.crosswalk['province'])[this.props.adminChildren.parent])];
-    const adminData = (level === 'province') ? Object.keys(this.props.VProMMsProps).filter(id => idTest.test(id.substring(0, 2))) : (Object.keys(this.props.VProMMsProps).filter(
-      (id) => {
-        id = id.substring(0, 5);
-        return idTest[0].test(id) && idTest[1].test(id);
-      })
-    );
-    const propertiesData = _.pickBy(this.props.VProMMsProps, (prop, vpromm) => (adminData.indexOf(vpromm) !== -1));
-    const field = (level === 'province') ? this.props.fieldIds.filter(vpromm => idTest.test(vpromm.substring(0, 2))).length : (Object.keys(this.props.VProMMsProps).filter(
-      (id) => {
-        id = id.substring(0, 5);
-        return idTest[0].test(id) && idTest[1].test(id);
-      }).length
+  componentWillReceiveProps: function (nextProps) {
+    // if the adminInfo is about to be fetched and ready for render
+    // grab the admin properties and field data needed to fill out the tables
+    if (this.props.adminInfoFetching && nextProps.adminInfoFetched) {
+      return this.getAdminData(nextProps);
+    }
+    // if switching between a district to its parent province,
+    // fetch that parent admin info.
+    if (this.props.location.pathname !== nextProps.location.pathname) {
```

Comment:
This check leads to an error when moving from `/analytics` to `/analytics/:aaid`.
![image](https://user-images.githubusercontent.com/1090606/31383341-9f3ed896-adb2-11e7-83ae-12d6b445a276.png)
This happens because the `pathname` did change, but the `adminInfo` is not fetched.

I don't quite understand what you're trying to do here. Why use the `parent.id` when changing page? If you're moving from district to province won't the URL reflect that change? Wouldn't it be enough to fetch data for the new iD?
```
    if (this.props.params.aaId !== nextProps.params.aaId) {
      return this.props._fetchAdminInfo(nextProps.params.aaId);
    }
```